services/business/checkers/virtual_services/single_host_checker.go

```diff
@@ -140,3 +142,13 @@ func formatHostForSearch(hostName, virtualServiceNamespace string) Host {
 
 	return host
 }
+
+func hasGateways(virtualService *kubernetes.IstioObject) bool {
+	if gateways, ok := (*virtualService).GetSpec()["gateways"]; ok {
+		if vsGateways, ok := (gateways).([]interface{}); ok {
+			return ok && vsGateways != nil && len(vsGateways) > 0
```

Comment:
``return ok`` here is kinda silly as this code isn't reached if it isn't true. Thus the real executed code is only ``return vsGateways != nil  && len(vsGateways) > 0"

Reply:
The other option of course would be to ditch the below ``return false`` and the "if clause" and return a combination of what you found.